src/mapgen.cpp

```diff
@@ -835,33 +835,38 @@ class jmapgen_item_group : public jmapgen_piece {
     }
 };
 
-/**
- * Place items from an item group.
- * see @ref map::place_loot
- */
+/** Place items from an item group */
 class jmapgen_loot : public jmapgen_piece {
+    friend jmapgen_objects;
+
     public:
         jmapgen_loot( JsonObject &jsi ) : jmapgen_piece()
         , group( jsi.get_string( "group", std::string() ) )
         , name( jsi.get_string( "item", std::string() ) )
-        , chance( jsi, "chance", 100, 100 )
-        , ammo( jsi, "ammo", 0, 0 )
-        , magazine( jsi, "magazine", 0, 0 )
+        , chance( jsi.get_int( "chance", 100 ) )
+        , ammo( jsi.get_int( "ammo", 0 ) )
+        , magazine( jsi.get_int( "magazine", 0 ) )
         {
             if( group.empty() == name.empty() ) {
                 jsi.throw_error( "must provide either item or group" );
             }
-
             if( !group.empty() && !item_group::group_is_defined( group ) ) {
                 jsi.throw_error( "no such item group", "group" );
             }
             if( !name.empty() && !item_controller->has_template( name ) ) {
                 jsi.throw_error( "no such item", "item" );
             }
+            if( ammo < 0 || ammo > 100 ) {
+                jsi.throw_error( "ammo chance out of range", "ammo" );
+            }
+            if( magazine < 0 || magazine > 100 ) {
+                jsi.throw_error( "ammo chance out of range", "ammo" );
```

Comment:
This message is absolutely the same as above, it's probably should mention "magazine" instead of "ammo".